src-tauri/src/plugin_management/download.rs

```diff
@@ -0,0 +1,162 @@
+use app::GameInstall;
+use once_cell::sync::OnceCell;
+use std::{
+    fs::{self, File, OpenOptions},
+    io,
+    path::PathBuf,
+    str::FromStr,
+};
+use tauri::{
+    async_runtime::{block_on, channel, Mutex, Receiver, Sender},
+    Manager, State,
+};
+use thermite::{core::utils::TempDir, prelude::ThermiteError};
+use zip::ZipArchive;
+
+use crate::{mod_management::ParsedThunderstoreModString, APP_HANDLE};
+
+static INSTALL_STATUS_RECV: OnceCell<Mutex<Receiver<bool>>> = OnceCell::new();
+
+pub struct InstallStatusSender(Mutex<Sender<bool>>);
+
+impl InstallStatusSender {
+    pub fn new() -> Self {
+        let (send, recv) = channel(1);
+
+        INSTALL_STATUS_RECV
+            .set(Mutex::new(recv))
+            .expect("failed to set INSTALL_STATUS_RECV");
+
+        Self(Mutex::new(send))
+    }
+}
+
+/// Tries to install plugins from a thunderstore zip
+pub async fn install_plugin(
+    game_install: &GameInstall,
+    zip_file: &File,
+    thunderstore_mod_string: &str,
+    can_install_plugins: bool,
+) -> Result<(), ThermiteError> {
+    let plugins_directory = PathBuf::new()
+        .join(&game_install.game_path)
+        .join("R2Northstar")
+        .join("plugins");
+    let temp_dir = TempDir::create(plugins_directory.join("___flightcore-temp-plugin-dir"))?;
```

Comment:
Any reason you create a second temp dir instead of using the existing one? ^^

Reply:
to unzip the zip

Reply:
Yeah but like couldn't you do that in the existing temp dir?

Reply:
[`___flightcore-temp-download-dir/`](https://github.com/R2NorthstarTools/FlightCore/pull/267/files#diff-e8d845933a9237c520ef224431b15bb0d5c9870431d41f06625e2663206371c3R372) should still exist while plugin is being installed ^^

Reply:
that's not a temp dir, it never gets cleaned up

Reply:
> it doesn't when an error is returned

Which should probably get fixed...

>  and also the mod installing code also uses its own temp dir.

Yeah, there's also a bunch of inconsistencies that need to be addressed (and are addressed on `libthermite` version bump IIRC).


Basically, if you can make it work, use existing temp dir, if it doesn't work, make a comment note that that's a temporary solution cause in the long run we only want a single temp dir so that it's easy to clean up by hand on errors etc

Reply:
Actually, you could just stick your temp dir into the other one as a subdir with little to no modification on your code needed, no?

Reply:
ok, this will be temporary

Reply:
This is still open ig

Reply:
I can wait until plugins v3